src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [a, b, c];
+    const sumOfSides = sides.reduce((acc, curr) => acc + curr, 0);
+    const maxSide = Math.max(...sides);
+
+    if (sides.some((side) => side <= 0)) {
+      throw new Error('One of the sides is less than 0.');
+    }
+
+    if (maxSide >= (sumOfSides - maxSide)) {
+      // eslint-disable-next-line
+      throw new Error('The longest side of a triangle is greater than or equal to the sum of the sides.');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const { a, b, c } = this;
+    const semiPerimeter = (a + b + c) / 2;
+    const square = Math.sqrt(
+      semiPerimeter
+      * (semiPerimeter - a)
+      * (semiPerimeter - b)
+      * (semiPerimeter - c),
+    );
 
+    return Number(square.toFixed(2));
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
 
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('The radius is less than or equal to 0');
+    }
+  }
+
+  getArea(): number {
+    return Math.floor((Math.PI * this.radius ** 2) * 100) / 100;
```

Comment:
make a separate function & reuse in all cases
also, split this with variable
```suggestion
    return getRounding(...);
```